src/Compilers/CSharp/Test/Semantic/Semantics/DeconstructionTests.cs

```diff
@@ -2546,5 +2546,25 @@ static void M((int, int) t)
             }
         }
 
+        [Fact, WorkItem(14287, "https://github.com/dotnet/roslyn/issues/14287")]
+        public void TupleDeconstructionStatementCannotBeConst()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        const (int x, int y) = (1, 2);
```

Comment:
There are two primary syntax forms for deconstruction:

``` csharp
(int x, int y) = ...
var (x, y) = ...
```

Consider adding a test for the latter form as well.